sources/api/early-boot-config/src/provider.rs

```diff
@@ -2,14 +2,18 @@
 
 use crate::settings::SettingsJson;
 
-#[cfg(any(bottlerocket_platform = "aws", bottlerocket_platform = "aws-dev"))]
-pub(crate) mod aws;
-
 #[cfg(bottlerocket_platform = "aws-dev")]
-pub(crate) mod local_file;
+mod local_file;
```

Comment:
I'm confused as to why local_file still has the cfg directive, since the PR says "It also makes local_file a common function that all platforms can use" - wouldn't we remove all conditionals on it, and let the data providers / platforms decide whether to call the function?

Reply:
We could for sure do that, but when compiling variants that don't use the code in the `local_file` module, we're going to get "unused" warnings like:
```
warning: constant is never used: `USER_DATA_FILE`
...
warning: function is never used: `local_file_user_data`
```

Reply:
I'm not really sure how we get around it other than hiding the whole module behind a `cfg` directive like I've done here... perhaps someone knows a trick I don't know?

Reply:
You could just silence the warning with `#[allow(dead_code)]` <- probably need to look up the correct string there.

Reply:
Hm - I suppose we could do that but I like it a bit less.  Are those really the only two options here? :thinking:

Reply:
[This (very good) article](https://bitshifter.github.io/2020/05/07/conditional-compilation-in-rust/) suggests the pattern I'm using: putting `cfg` directives on modules to keep the actual code clean and not ignore any valid warnings.

IMHO let's keep the code as is unless we come up with something better.